### PR TITLE
Document HttpServerOptionsCustomizer

### LINKED ISSUE #34992: Document usage of `HttpServerOptionsCustomizer`
### Description

Document usage of `HttpServerOptionsCustomizer`

Quarkus docs do not mention `HttpServerOptionsCustomizer`, we have it just in tests for vertx-http module

CC @cescoffier 

```
git grep HttpServerOptionsCustomizer

extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java:import io.quarkus.vertx.http.HttpServerOptionsCustomizer;
extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java:    UnremovableBeanBuildItem shouldNotRemoveHttpServerOptionsCustomizers() {
extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java:        return UnremovableBeanBuildItem.beanTypes(HttpServerOptionsCustomizer.class);
extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/customizers/HttpServerOptionsCustomizerTest.java:import io.quarkus.vertx.http.HttpServerOptionsCustomizer;
extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/customizers/HttpServerOptionsCustomizerTest.java:public class HttpServerOptionsCustomizerTest {
extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/customizers/HttpServerOptionsCustomizerTest.java:    public static class MyCustomizer implements HttpServerOptionsCustomizer {
extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/HttpServerOptionsCustomizer.java:public interface HttpServerOptionsCustomizer {
extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:import io.quarkus.vertx.http.HttpServerOptionsCustomizer;
extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:            List<InstanceHandle<HttpServerOptionsCustomizer>> instances = Arc.container()
extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:                    .listAll(HttpServerOptionsCustomizer.class);
extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:            for (InstanceHandle<HttpServerOptionsCustomizer> instance : instances) {
extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:                HttpServerOptionsCustomizer customizer = instance.get();
```

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #34992

### CODE REVIEW

File: docs/src/main/asciidoc/http-reference.adoc [hunk 1]
```diff
@@ -442,6 +442,27 @@ include::{generated-dir}/config/quarkus-vertx-http-config-group-access-log-confi
 Use `quarkus.http.access-log.exclude-pattern=/some/path/.*` to exclude all entries concerning the path `/some/path/...` (_including subsequent paths_) from the log.
 ====
 
+== Arbitrary customizations
+
+Quarkus allows users to arbitrarily customize the options of HTTP server(s) that is(are) started by Quarkus via the use of `io.quarkus.vertx.http.HttpServerOptionsCustomizer`.
+For example, the HTTP port needs to be set programmatically, then the following code could be used:
```
Comment (gsmet) [inline docs/src/main/asciidoc/http-reference.adoc:448]:
```suggestion
For example, if the HTTP port needs to be set programmatically, then the following code could be used:
```

**Review by gsmet — DISMISSED** [review]

File: docs/src/main/asciidoc/http-reference.adoc [hunk 2]
```diff
@@ -442,6 +442,27 @@ include::{generated-dir}/config/quarkus-vertx-http-config-group-access-log-confi
 Use `quarkus.http.access-log.exclude-pattern=/some/path/.*` to exclude all entries concerning the path `/some/path/...` (_including subsequent paths_) from the log.
 ====
 
+== Arbitrary customizations
+
+Quarkus allows users to arbitrarily customize the options of HTTP server(s) that is(are) started by Quarkus via the use of `io.quarkus.vertx.http.HttpServerOptionsCustomizer`.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/http-reference.adoc:447]:
```suggestion
Quarkus allows users to arbitrarily customize the options of HTTP servers started by Quarkus via the use of `io.quarkus.vertx.http.HttpServerOptionsCustomizer`.
```

File: docs/src/main/asciidoc/http-reference.adoc [hunk 3]
```diff
@@ -442,6 +442,27 @@ include::{generated-dir}/config/quarkus-vertx-http-config-group-access-log-confi
...
+public static class MyCustomizer implements HttpServerOptionsCustomizer {
+
+    @Override
+    public void customizeHttpServer(HttpServerOptions options) { <2>
+        options.setPort(9998);
+    }
+}
+----
+<1> By making the class a bean, Quarkus will take the customizer into account when it starts the Vert.x servers
+<2> We only care about customizing the HTTP server in this case, so we only override the `customizeHttpServer` method.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/http-reference.adoc:465]:
```suggestion
<2> In this case, we only care about customizing the main HTTP server, so we just override the `customizeHttpServer` method.
```

**Review by gastaldi — APPROVED** [review]
LGTM. Rephrased some sentences for clarity purposes

File: docs/src/main/asciidoc/http-reference.adoc [hunk 4]
```diff
@@ -442,6 +442,27 @@ include::{generated-dir}/config/quarkus-vertx-http-config-group-access-log-confi
...
+@Singleton <1>
+public static class MyCustomizer implements HttpServerOptionsCustomizer {
+
+    @Override
+    public void customizeHttpServer(HttpServerOptions options) { <2>
+        options.setPort(9998);
+    }
+}
+----
+<1> By making the class a bean, Quarkus will take the customizer into account when it starts the Vert.x servers
```
Comment (gastaldi) [inline docs/src/main/asciidoc/http-reference.adoc:464]:
```suggestion
<1> By making the class a managed bean, Quarkus will take the customizer into account when it starts the Vert.x servers
```

**Review by cescoffier — CHANGES_REQUESTED** [review]
It should mention the other methods when using domain sockets and https.

**Comment by geoand** [comment]
> It should mention the other methods when using domain sockets and https.

Done

File: docs/src/main/asciidoc/http-reference.adoc [hunk 5]
```diff
@@ -442,6 +442,27 @@ include::{generated-dir}/config/quarkus-vertx-http-config-group-access-log-confi
 Use `quarkus.http.access-log.exclude-pattern=/some/path/.*` to exclude all entries concerning the path `/some/path/...` (_including subsequent paths_) from the log.
 ====
 
+== Arbitrary customizations
+
+Quarkus allows users to arbitrarily customize the options of HTTP servers started by Quarkus via the use of `io.quarkus.vertx.http.HttpServerOptionsCustomizer`.
+For example, if the HTTP port needs to be set programmatically, then the following code could be used:
```
Comment (rsvoboda) [inline docs/src/main/asciidoc/http-reference.adoc:448]: Can be everything from https://vertx.io/docs/apidocs/io/vertx/core/http/HttpServerOptions.html configured or there are some limitations that Quarkus imposes?

Comment (geoand) [reply docs/src/main/asciidoc/http-reference.adoc:448]: What kind of limitations are you thinking of? 

There isn't anything I am personally aware of

Comment (geoand) [reply docs/src/main/asciidoc/http-reference.adoc:448]: That's pretty much implied, no?

Comment (cescoffier) [reply docs/src/main/asciidoc/http-reference.adoc:448]: No limitations, but also... no safety belt...

**Comment by geoand** [comment]
We actually have people already using [this](https://github.com/quarkusio/quarkus/issues/15926#issuecomment-1545002471)

**Review by cescoffier — APPROVED** [review]

**Pull request merged**